project/src/main/java/com/google/autograder/servlets/helpers/API.java

```diff
@@ -33,27 +31,25 @@
 
 public final class API {
 
-    private static final String RESOURCES_PATH  = "../../../project/src/main/resources"; // "../classes"
     private static final Logger LOGGER = Logger.getLogger(API.class.getName());
-    private static final String API_KEY_PATH = "/auth/api_key.json";
-    private static final Gson GSON = new Gson();
+    private static final String API_KEY_PATH = "auth/api_key.json";
 
     public static final String UTF_8 = StandardCharsets.UTF_8.name();
     public static final String API_KEY = retrieveAPIKey().get();
 
     private static Optional<String> retrieveAPIKey() {
```

Comment:
Similar to the other PR, throw a RuntimeException here rather than returning an Optional. It'll have the same effect but will consolidate all the logic in one place

Reply:
+1